### PR TITLE
Status.im theme

### PR DESCRIPTION
**Pull request by ara4n**
Land the long-lived Status.im theme branch. Obviously this should ideally be split up into PRs, but unfortunately life is too short atm :/

Twinned with https://github.com/matrix-org/matrix-react-sdk/pull/1605

Stuff to be aware of here:
 * Completely refactors and consolidates the CSS for buttons (using a SCSS mixin rather than an CSS class)
 * Complete rewrites the Tinter to handle tintable themes
 * Might introduce regressions into the default Riot theme where refactoring happened in order to incorporate the Status theme.

### CODE REVIEW

File: config.sample.json [hunk 1]
```diff
@@ -1,6 +1,10 @@
 {
     "default_hs_url": "https://matrix.org",
     "default_is_url": "https://vector.im",
+    "disable_custom_urls": false,
+    "disable_guests": false,
+    "disable_login_language_selector": false,
+    "disable_3pid_login": false,
```
Comment (turt2live) [inline config.sample.json]: Note to self: These should be pulled into GranularSettings when it lands

File: docs/skinning thoughts.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  - gather together themes (per skin) into a single place too
+    - skins/vector/themes/foo/css
+    - skins/vector/themes/foo/img
+    - skins/vector/themes/foo/fonts
+  - ideally riot-web would contain almost nothing but skins/vector directory. 
+  - ability to entirely replace CSS rather than override it for a given theme
+    - e.g. if we replace `Login.js` with `StatusLogin.js`, then we should similarly be able to replace `_Login.scss` with `_StatusLogin.scss`.
+
+random thoughts;
+   - should we be able to change the entire skin at runtime (more like wordpress) - to the extent of replacing entire components?
```
Comment (MTRNord) [inline docs/skinning thoughts.md]: What about color pickers for now? Not like just tint but changing different parts of the Design without needing to change the code

File: src/vector/index.js [hunk 3]
```diff
@@ -250,16 +251,20 @@ async function loadApp() {
 
     if (!preventRedirect) {
         if (/iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream) {
-            if (confirm(_t("Riot is not supported on mobile web. Install the app?"))) {
+            window.location = "https://status.im/join-riot.html";
```
Comment (MTRNord) [inline src/vector/index.js]: Won't that cause problems for selfhosted Riots?

Comment (betz) [reply src/vector/index.js]: thanks for noticing this.
Yes, please don't hijack people

Comment (ara4n) [reply src/vector/index.js]: heh, looks like this crept in whilst my back was turned. of course it’s not meant to be there; will turn it into a config options. thanks!

Comment (MTRNord) [reply src/vector/index.js]: @ara4n Note a few lines below this is the same line again ;)

**Comment by ara4n** [comment]
@dbkr LGTMed this at high level OOB, so merging.  It'll probably break everything, but y'all should be testing /staging for 0.13.0-rc2 anyway, and I want to give @turt2live a chance to merge it to his GranularSettings tonight (and then merge that too).

File: config.sample.json (hunk 1 above)
Comment (turt2live) [reply config.sample.json]: decided to not make these granular - they are app-level anyways and affect people who aren't logged in, so granularity isn't feasible.

**Pull request merged**